shared/src/models/utils/occupations/occupation-helpers-mutable.ts

```diff
@@ -0,0 +1,28 @@
+import type { Mutable } from '../../../utils';
+import { cloneDeepMutable } from '../../../utils';
+import type { ExerciseOccupation } from './exercise-occupation';
+import { NoOccupation } from './no-occupation';
+
+export function isUnoccupied(
+    occupied: Mutable<{ occupation: ExerciseOccupation }>,
+    currentTime: number
+) {
+    if (
+        occupied.occupation.type === 'intermediateOccupation' &&
+        occupied.occupation.unoccupiedUntil < currentTime
+    ) {
+        occupied.occupation = cloneDeepMutable(NoOccupation.create());
+    }
+
+    return occupied.occupation.type === 'noOccupation';
+}
+
+export function isUnoccupiedOrIntermediaryOccupied(
```

Comment:
I have not found the adverb in any dictionary but something like `intermediarily` should get the point across.

Reply:
2b95c61fffc46093fbdc363bb2b0e79f5b12b967